### PR TITLE
add: built in api support is added, as an example GET /api/health-check endpoint is added

### PR DESCRIPTION
**Pull request by mehmetsefabalik**

### CODE REVIEW

**Comment by omeraplak** [comment]
Hi @mehmetsefabalik ,
Could you add feature usage documentation?

File: pages/api/health-check.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default (req: NextApiRequest, res: NextApiResponse) => {
```
Comment (omeraplak) [inline pages/api/health-check.ts:3]: Could you add return typedef?

**Comment by mehmetsefabalik** [comment]
> Hi @mehmetsefabalik ,
> Could you add feature usage documentation?

this is the feature of nextjs, it is well documented in nextjs docs. So the user of this boilerplate is supposed to read this feature from nextjs docs.

File: pages/api/health-check.ts (hunk 1 above)
Comment (mehmetsefabalik) [reply pages/api/health-check.ts]: return typedef is any, I added it

File: pages/api/health-check.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default (req: NextApiRequest, res: NextApiResponse): any => {
+    res.status(200).json({ status: "working" });
```
Comment (BatuhanW) [inline pages/api/health-check.ts:4]: Let's make this 'OK', instead of 'working'.

**Comment by BatuhanW** [comment]
> > Hi @mehmetsefabalik ,
> > Could you add feature usage documentation?
> 
> this is the feature of nextjs, it is well documented in nextjs docs. So the user of this boilerplate is supposed to read this feature from nextjs docs.

We can still point to related documentation links 👍

File: pages/api/health-check.ts (hunk 2 above)
Comment (mehmetsefabalik) [reply pages/api/health-check.ts]: done

**Comment by mehmetsefabalik** [comment]
> > > Hi @mehmetsefabalik ,
> > > Could you add feature usage documentation?
> > 
> > 
> > this is the feature of nextjs, it is well documented in nextjs docs. So the user of this boilerplate is supposed to read this feature from nextjs docs.
> 
> We can still point to related documentation links 👍

done

**Comment by aliemir** [comment]
Hi @mehmetsefabalik, thank you for your contribution. I noticed that only get requests are handled by the custom server side. Since you are adding the api routes support, can you check the server files to handle other http methods?

[server/index.ts#L34](https://github.com/pankod/next-boilerplate/blob/master/server/index.ts#L34)

I'm guessing changing this line from `.get()` to `.all()` might be enough.

**Comment by mehmetsefabalik** [comment]
> Hi @mehmetsefabalik, thank you for your contribution. I noticed that only get requests are handled by the custom server side. Since you are adding the api routes support, can you check the server files to handle other http methods?
> 
> [server/index.ts#L34](https://github.com/pankod/next-boilerplate/blob/master/server/index.ts#L34)
> 
> I'm guessing changing this line from `.get()` to `.all()` might be enough.

done

File: pages/api/health-check.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default (req: NextApiRequest, res: NextApiResponse): any => {
```
Comment (omeraplak) [inline pages/api/health-check.ts:3]:
```suggestion
export default (req: NextApiRequest, res: NextApiResponse): void => {
```

**Comment by omeraplak** [comment]
Thank you for contribution @mehmetsefabalik

**Pull request merged**